portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,19 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+/**
+ * Async function loads the server content
+ */
+async function randomFactLoadFromServer(){
+    const serverResponse = await fetch('/facts');
+    const serverResponseJson = await serverResponse.json();
```

Comment:
What happens in this function if `await fetch('/facts')` throws an IOException? You should be able to test it by going into your FactsServlet and in the doGet method, temporarily adding `throw new IOException();` 

I'm not 100% sure since I haven't done much web development, but it might be that we have to handle the use case where the `serverResponse` object isn't valid and therefore we can't request the `json()` from it!

Reply:
Thanks Gina and Sarah for pointing this out! I overlooked this very important part of testing. But I just tried to test it by adding a `throw new IOException()` in the `doGet` method in the `FactsServlet` file, but for some reason, the `catch` code block is not showing up on the console page. I will try to figure it out this part, and I will let you know my update on this!